diff --git a/src/gateways/stdioToSse.ts b/src/gateways/stdioToSse.ts
index 888e0e5..4c4801e 100644
--- a/src/gateways/stdioToSse.ts
+++ b/src/gateways/stdioToSse.ts
@@ -72,10 +72,7 @@ export async function stdioToSse(args: StdioToSseArgs) {
     process.exit(code ?? 1)
   })
 
-  const server = new Server(
-    { name: 'supergateway', version: getVersion() },
-    { capabilities: {} },
-  )
+
 
   const sessions: Record<
     string,
@@ -112,6 +109,10 @@ export async function stdioToSse(args: StdioToSseArgs) {
     })
 
     const sseTransport = new SSEServerTransport(`${baseUrl}${messagePath}`, res)
+    const server = new Server(
+      { name: 'supergateway', version: getVersion() },
+      { capabilities: {} },
+    )
     await server.connect(sseTransport)
 
     const sessionId = sseTransport.sessionId
